### PR TITLE
feat: implement push notification subscriptions

### PR DESCRIPTION
**Pull request by santi**
Implements storage of notification subscriptions for later use when sending notifications to friends

TODO: research the `auth` part of a subscription request. Is it safe to store it in plaintext, or should it be encrypted?

Note: PR is against the other branch which it builds on.

### CODE REVIEW

File: coffee-backend/src/main/kotlin/chat/letscoffee/notification/NotificationController.kt [hunk 1]
```diff
@@ -16,29 +16,55 @@ import org.springframework.web.bind.annotation.RestController
 @RequestMapping("/notifications")
 class NotificationController(
     private val notificationService: NotificationService,
+    private val notificationSubscriptionRepository: NotificationSubscriptionRepository,
```
Comment (vhellem) [inline coffee-backend/src/main/kotlin/chat/letscoffee/notification/NotificationController.kt:19]: Considering the other refactoring that has been done in these branches, why is this controller given direct access to the repository. Should move Should ask the service for this?

File: coffee-backend/src/main/kotlin/chat/letscoffee/notification/NotificationController.kt [hunk 2]
```diff
@@ -16,29 +16,55 @@ import org.springframework.web.bind.annotation.RestController
...
+        @RequestBody subscriptionRequest: SubscriptionRequest
     ): ResponseEntity<String> {
+        val user = userService.getUserById(userPrincipal.id)
 
-        // TODO: Save user notification subscription for later use
-        val response = notificationService.sendPushNotification(notificationSubscription)
-        // TODO: Handle response from PUSH API server gracefully
-
-        return ResponseEntity("Successfully subscribed to push notifications", HttpStatus.ACCEPTED)
+        if (notificationSubscriptionRepository.existsByUserId(user)) {
```
Comment (vhellem) [inline coffee-backend/src/main/kotlin/chat/letscoffee/notification/NotificationController.kt:31]: Ask service?

File: coffee-backend/src/main/kotlin/chat/letscoffee/notification/NotificationController.kt (hunk 1 above)
Comment (santi) [reply coffee-backend/src/main/kotlin/chat/letscoffee/notification/NotificationController.kt:19]: Agreed. Didn't want to touch all the services/repos at the same time. Will refactor in later PR :+1:

File: coffee-backend/src/main/kotlin/chat/letscoffee/notification/NotificationController.kt (hunk 2 above)
Comment (santi) [reply coffee-backend/src/main/kotlin/chat/letscoffee/notification/NotificationController.kt:31]: Will add in next PR 🍎

**Pull request merged**